libcxx/test/support/test_macros.h

```diff
@@ -214,10 +214,9 @@
 #define TEST_IS_EXECUTED_IN_A_SLOW_ENVIRONMENT
 #endif
 
-#if defined(_LIBCPP_HAS_NO_ALIGNED_ALLOCATION) || \
-  (!(TEST_STD_VER > 14 || \
-    (defined(__cpp_aligned_new) && __cpp_aligned_new >= 201606L)))
-#define TEST_HAS_NO_ALIGNED_ALLOCATION
+#if (!defined(_LIBCPP_HAS_ALIGNED_ALLOCATION) || !_LIBCPP_HAS_ALIGNED_ALLOCATION) ||                                   \
+    (!(TEST_STD_VER > 14 || (defined(__cpp_aligned_new) && __cpp_aligned_new >= 201606L)))
+#  define TEST_HAS_NO_ALIGNED_ALLOCATION
```

Comment:
I think I would split this a bit to make it easier to understand. Also, I think it's a bit wrong how it was refactored. Instead:

```
#if defined(_LIBCPP_VERSION) && !_LIBCPP_HAS_ALIGNED_ALLOCATION
#  define TEST_HAS_NO_ALIGNED_ALLOCATION
#elif TEST_STD_VER < 17 && (!defined(__cpp_aligned_new) || __cpp_aligned_new < 201606L)
#  define TEST_HAS_NO_ALIGNED_ALLOCATION
#endif
```